core/src/scene/sceneLoader.cpp

```diff
@@ -520,36 +520,36 @@ bool SceneLoader::extractTexFiltering(Node& filtering, TextureFiltering& filter)
 }
 
 void SceneLoader::updateSpriteNodes(const std::string& texName,
-        std::shared_ptr<Texture>& texture, Scene& scene) {
-    auto& spriteAtlases = scene.spriteAtlases();
+        std::shared_ptr<Texture>& texture, const std::shared_ptr<Scene>& scene) {
+    auto& spriteAtlases = scene->spriteAtlases();
     if (spriteAtlases.find(texName) != spriteAtlases.end()) {
         auto& spriteAtlas = spriteAtlases[texName];
         spriteAtlas->updateSpriteNodes(texture);
     }
 }
 
 std::shared_ptr<Texture> SceneLoader::fetchTexture(const std::string& name, const std::string& url,
-        const TextureOptions& options, bool generateMipmaps, Scene& scene) {
+        const TextureOptions& options, bool generateMipmaps, const std::shared_ptr<Scene>& scene) {
 
     std::shared_ptr<Texture> texture;
 
     std::regex r("^(http|https):/");
     std::smatch match;
     // TODO: generalize using URI handlers
     if (std::regex_search(url, match, r)) {
-        scene.m_resourceLoad++;
-        startUrlRequest(url, [=, &scene](std::vector<char>&& rawData) {
+        scene->m_resourceLoad++;
+        startUrlRequest(url, [=](std::vector<char>&& rawData) {
                 auto ptr = (unsigned char*)(rawData.data());
```

Comment:
Yes and hence this: https://github.com/tangrams/tangram-es/issues/814

On Jul 18, 2016 6:57 AM, "Hannes Janetzek" <notifications@github.com> wrote:

> In core/src/scene/sceneLoader.cpp
> <https://github.com/tangrams/tangram-es/pull/837#discussion_r71129375>:
>
> >
> >      std::shared_ptr<Texture> texture;
> >
> >      std::regex r("^(http|https):/");
> >      std::smatch match;
> >      // TODO: generalize using URI handlers
> >      if (std::regex_search(url, match, r)) {
> > -        scene.m_resourceLoad++;
> > -        startUrlRequest(url, [=, &scene](std::vector<char>&& rawData) {
> > +        scene->m_resourceLoad++;
> > +        startUrlRequest(url, [=](std::vector<char>&& rawData) {
> >                  auto ptr = (unsigned char*)(rawData.data());
>
> Ah - I see the issue. Shouldn't scene-loading block and wait until all
> resources are available?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/837/files/29a9b8eb1536306a6eb2f7df96bf99a3f6449246#r71129375>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAWAwTXHVCU_DOu2fP7BmfzCYyv_JYnTks5qW1wMgaJpZM4JMxQx>
> .
>